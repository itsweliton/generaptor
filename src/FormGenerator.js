import React from 'react';
import './App.css';

const FormGenerator = ({ schema }) => {
  if (!schema || !schema.fields) throw new Error('You must have a schema');


  return (
    <div>
      <form>
        {schema.fields.map(field => (
          <div key={field.id}>
              <label for="name">{field.label}</label>
              <input type="text" id={field.id} name="name" required minlength="4" maxlength="8" size="10" placeholder={field.placeholder}></input>
          </div>)
        )}
      </form>
    </div>
  );
}

export default FormGenerator;
