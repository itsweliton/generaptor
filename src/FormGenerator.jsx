import React from "react";
import FormBuilder from "./FormBuilder";
import "./App.css";

const FormGenerator = function () {
  return {
    create: (schema, options) => {
      if (!schema || schema === undefined) return;

      return <FormBuilder fields={schema.fields} options={options} />;
    },
  };
};

export default FormGenerator;
