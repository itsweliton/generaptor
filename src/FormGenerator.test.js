import React from 'react';
import ReactDOM from 'react-dom';
import FormGenerator from './FormGenerator';
import { shallow, mount } from 'enzyme';
import { create } from 'react-test-renderer';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const schema = { 
  fields: [
    {
      id: "le",
      type: "input",
      inputType: "text",
      place: "Name",
      placeholder: "Name",
      model: "name"
    }
  ]
}

// describe("Form Generator Mounted", () => {
//   it('Exports a valid component', () => {
//     const div = document.createElement('div');
//     ReactDOM.render(<FormGenerator schema={{fields: []}}/>, div);
//     ReactDOM.unmountComponentAtNode(div);
//   });

//   test("Component is created", () => {
//     const formGenerator = create(<FormGenerator schema={{fields: []}}/>);
//     expect(formGenerator.toJSON()).toMatchSnapshot();
//   });

//   it("Component without schema throws error", () => {
//     expect(() => mount(<FormGenerator />)).toThrowError(/^You must have a schema$/);
//   });
// })

describe("Form Generator basic mounting", () => {
  test("It should create a basic input", () => {
    const wrapper = shallow(<FormGenerator schema={schema} />);
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(`input#le`).props().placeholder).toBe("Name");
  });
});