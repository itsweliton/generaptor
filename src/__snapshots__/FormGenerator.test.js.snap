// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Generator basic mounting It should create a basic input 1`] = `ShallowWrapper {}`;
